plugins/capture.rb

```diff
@@ -0,0 +1,605 @@
+      File.join(Msf::Config.log_directory,"captures/#{name}")
+    end
+
+    def default_hashdir(options)
+      session = 'local'
+      session = options[:session].to_s unless options[:session].nil?
+
+      name = "capture_#{session}_#{Time.now.strftime('%Y%m%d%H%M%S')}_#{Rex::Text.rand_text_numeric(6)}"
+      File.join(Msf::Config.loot_directory,"captures/#{name}")
+    end
+
+    def read_config(filename)
+      options = {}
+      File.open(filename, "rb") do |f|                                       
+        yamlconf = YAML::load(f)
+        options = {
+         :ntlm_challenge => yamlconf['ntlm_challenge'],
+         :ntlm_domain => yamlconf['ntlm_domain'],
+         :services => yamlconf['services']
+        }
+      end
+    end
+    
+    def parse_stop_args(args)
+      options = {
+        :session => nil,
+        :show_help => false,
+      }
+
+      @@start_opt_parser.parse(args) do |opt, idx, val|
+        case opt
+        when '--session'
+          options[:session] = val
+        when '-h'
+          options[:show_help] = true
+        end
+      end
+
+      options
+    end
+
+    def parse_start_args(args)
+      config_file = File.join(Msf::Config.data_directory,"capture_config.yaml")
+      # See if there was a config file set
+      @@start_opt_parser.parse(args) do |opt, idx, val|
+        case opt
+        when '--configfile'
+          config_file = val
+        end
+      end
+
+      options = default_options
+      config_options = read_config(config_file)
+      options = options.merge(config_options)
+
+      @@start_opt_parser.parse(args) do |opt, idx, val|
+        case opt
+        when '--session'
+          options[:session] = val
+        when '-i'
+          options[:srvhost] = val
+        when '--spoofip'
+          options[:spoof_ip] = val
+        when '--regex'
+          options[:spoof_regex] = val
+        when '-v'
+          options[:verbose] = true
+        when '--basic'
+          options[:http_basic] = true
+        when '--cert'
+          options[:ssl_cert] = val
+        when '--stdout'
+          options[:stdout] = true
+        when '--logfile'
+          options[:logfile] = val
+        when '--hashdir'
+          options[:hashdir] = val
+        when '-h'
+          options[:show_help] = true
+        end
+      end
+
+      options
+    end
+
+    def poison_included(options)
+      poisoners = ['mDNS','LLMNR','NBNS']
+      poisoners.each do |svc|
+        if options[:services][svc]
+          return true
+        end
+      end
+      false
+    end
+
+    # Fill in implied parameters to make the running code neater
+    def transform_params(options)
+      # If we've been given a specific IP to listen on, use that as our poisoning IP
+      if options[:spoof_ip].nil? && options[:srvhost] != '0.0.0.0'
+        options[:spoof_ip] = options[:srvhost]
+      end
+      
+      unless options[:session].nil?
```

Comment:
Before using the session number, it'd be a good idea to resolve it in case the user specified `-1`.

```suggestion
      unless options[:session].nil?
        options[:session] = framework.session.get(options[:session])&.sid
```